Test/HUnit/Lang.hs

```diff
@@ -89,4 +91,8 @@ performTestCase action =
        -- is not the case, please email the maintainer.
        E.Handler (\e -> throw (e :: E.AsyncException)),
 
+#if MIN_VERSION_base(4,9,0)
+       E.Handler (\(E.ErrorCallWithLocation msg _) -> return $ Error Nothing msg),
```

Comment:
There's a cost involved with attaching stack-traces/location infos everywhere; for one, here's how the type-signatures now look like in GHCi 8.0:

```hs
λ:1> :t error
error :: forall (v :: GHC.Types.Levity) (a :: TYPE v). (?callStack::GHC.Stack.Types.CallStack) => [Char] -> a
λ:2> :t undefined 
undefined :: forall (v :: GHC.Types.Levity) (a :: TYPE v). (?callStack::GHC.Stack.Types.CallStack) => a
```

So it was decided to limit the scope only to `error`/`undefined` for the meantime. I've forgotten the details, but I can try to find and link you the original discussion if you're interested.

The reason I had to add that line was to have the testsuite pass, and have a consistent behaviour across GHC versions (otherwise sometimes you'd get a mangled error-message, and sometimes not depending on your GHC version). I'd suggest extending the `Error`-type to include an additional field rather than merging together the callstack-info-string with the original error message (we added a 2nd field into the `ErrorCall` record for the very purpose of being able to retrieve the original unaltered message)